### PR TITLE
Add a select box to load the agreement in different languages while r…

### LINKED ISSUE #2892: Checkbox agreement should allow for language selection

### PR DESCRIPTION
**Pull request by emanuele45**
…egistering - fixes #2892

### CODE REVIEW

**Comment by emanuele45** [comment]
Okay, I'm a little lost here. :laughing:

**Comment by Spuds** [comment]
You mean the Travis 5.6 error ?   Something changed when they began the Trusty rollout.  I had it all working just fine a week before this on the Trusty environment.  I'm frankly in no mood to debug that ATM, getting it to work at all is a dependency version level nightmare.

**Comment by emanuele45** [comment]
I tried changing some versions, but I didn't get anything.
I'll revert the changes and give up for the moment.

**Comment by Spuds** [comment]
Believe me I understand the frustration with getting it to run.   It took quite a bit of effort to get the combination that WAS working with trusty .. the number of incompatibles with different drivers and browsers and php unit selenium is crazy stupid.

File: themes/default/scripts/register.js [hunk 1]
```diff
@@ -414,4 +414,30 @@ function onCheckChange()
...
+					$('#agreement_box').html(request);
+				}
+			})
+			.fail(function(request) {
+				// Maybe reload with language attribute?
+			})
+			.always(function() {
+				// turn off the indicator
+				ajax_indicator(false);
+			})
```
Comment (Frenzie) [inline themes/default/scripts/register.js]: You forgot the semicolon here. :-P

File: themes/default/scripts/register.js [hunk 2]
```diff
@@ -414,4 +414,30 @@ function onCheckChange()
...
+	$(document).ready(function() {
+		$('#agreement_lang').change(function() {
+			$.ajax({
+				type: "POST",
+				url: elk_scripturl + "?action=register;xml;api=json;",
+				data: {sa: 'agrelang', lang: $(this).val()},
+				beforeSend: ajax_indicator(true)
+			})
+			.done(function(request) {
+				if (request != '')
```
Comment (Frenzie) [inline themes/default/scripts/register.js]: Is that safe or should it be something like `if (request && request != '')`?

File: sources/controllers/Register.controller.php [hunk 3]
```diff
@@ -1258,4 +1265,42 @@ public function action_registerCheckUsername()
 
 		$context['valid_username'] = !$errors->hasErrors();
 	}
+
+	public function action_agrelang()
```
Comment (Frenzie) [inline sources/controllers/Register.controller.php]: Why not `action_agreelang`? Does that already exist? Something against double vowels? :-P

File: themes/default/scripts/register.js (hunk 1 above)
Comment (emanuele45) [inline themes/default/scripts/register.js]: It's a foreseen exception. :-P

File: themes/default/scripts/register.js (hunk 2 above)
Comment (emanuele45) [inline themes/default/scripts/register.js]: hmm... I don't **think** request can be not set and it should at least be a string (though jquery documentation is not exactly easy to read).

File: sources/controllers/Register.controller.php (hunk 3 above)
Comment (emanuele45) [inline sources/controllers/Register.controller.php]: Yeah, I was considering action_agreementlang but it seemed a bit on the longish side.
agreelang looks indeed slightly better than agrelang.

Comment (emanuele45) [inline sources/controllers/Register.controller.php]: Actually, I realized there is a better place for localization-related stuff, so I moved it there. I had to move a bit of other stuff around, but I think it looks nicer now... maybe. :stuck_out_tongue:

**Pull request merged**